tinyagentos/projects/project_store.py

```
@@ -261,28 +274,28 @@ async def update_project(
             return
         sets.append("updated_at = ?"); params.append(time.time())
         params.append(project_id)
-        await self._db.execute(
-            f"UPDATE projects SET {', '.join(sets)} WHERE id = ?", params
-        )
-        await self._db.commit()
+        async with self._tx():
+            await self._db.execute(
+                f"UPDATE projects SET {', '.join(sets)} WHERE id = ?", params
```

Ruff (0.16.3)

[error] Possible SQL injection vector through string-based query construction

(S608)